### DISCUSSION TITLE
DietPi-Software | WSDD

### DISCUSSION THREAD

**Discussion by Joulinar** [MichaIng/DietPi#5244] [Software Request]
Hi,

quite a nice tool that could be installed in addition to SAMBA Server. This enables (Samba) hosts, like a local DietPi device, to be found by Web Service Discovery Clients like Windows.

Some background why this might be needed https://github.com/christgau/wsdd#background

Debian install is available via own apt repository https://github.com/christgau/wsdd#debianubuntu

**Comment by MichaIng** [comment]
Jep, sounds reasonable. Probably an option embedded with the Samba server install? Probably we can test this with recent Windows 10, verify that the server is not discovered without but with WSDD.

But it has to be noted that this _can_ be seen as security downside, similarly like people may prefer a hidden WiFi network. Also not sure how often people really use the network discovery instead of simply typing in the hostname or IP. But e.g. in combination with a dynamic IP (where required or wanted for some reason) convenient.

**Reply by Joulinar** [reply]
At least on my W11 laptop, it's working. DietPi is showing up within a second as soon as the service is started. And going to disappear once stopped. Basically the idea was build on a request on the forum where a user was missing DietPi on network overview on Win. But we could handle it same way as we do with unbound if Pihole or AGH has been selected and ask user if they like to activate the feature.

For security there some information on the GitHub pointing to the fact that the tool is mostly used in private network.

**Reply by MichaIng** [reply]
> But we could handle it same way as we do with unbound if Pihole or AGH has been selected and ask user if they like to activate the feature.

Since it is for the Samba server only (?), I think it makes sense like that.

**Comment by NossieUK** [comment]
This should really be an option on every distro =)

**Comment by frankk74** [comment]
I use this as well. It should be logged upstream against samba…

It seems they are considering it… 

https://bugzilla.samba.org/show_bug.cgi?id=4545

I’ve added the request upstream. I think this can be closed?

**Reply by Joulinar** [reply]
WSDD has become part of official Debian package repository. Can be installed via `apt `now 😉 

Probably the successor package https://packages.debian.org/bookworm/wsdd2

**Reply by MichaIng** [reply]
NIce, a C programmed alternative to the Python one. But both are dedicated projects, just implementing the same protocols with a different language. The "2" is misleading. The one from Steffen Christgau has seen its last commit more recently:
- https://github.com/christgau/wsdd
- https://github.com/Netgear/wsdd2

The C variant `wsdd2` should be way more lightweight. As I understand, there is not relevant interface or any such,it is just installing it, having a daemon running which does the WSD propagation automatically, and either it works, i.e. Windows sees the SMB server, or not, right? So if `wsdd2` just works as well, I would prefer it.

We could just add some prompt to the Samba install code to ask whether it shall be installed, and a related `dietpi.txt` option for automating this as well. I suggest to do so on Bookworm only to avoid adding any 3rd party repo, but that should not be an issue now that we switched all downloads over to Bookworm.

**Reply by Joulinar** [reply]
WSDD2 is working quite well

```
root@DietPiOPi5:~# apt install wsdd2
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
The following NEW packages will be installed:
  wsdd2
0 upgraded, 1 newly installed, 0 to remove and 0 not upgraded.
Need to get 25.4 kB of archives.
After this operation, 96.3 kB of additional disk space will be used.
Get:1 https://deb.debian.org/debian bookworm/main arm64 wsdd2 arm64 1.8.7+dfsg-1.1 [25.4 kB]
Fetched 25.4 kB in 0s (173 kB/s)
debconf: delaying package configuration, since apt-utils is not installed
Selecting previously unselected package wsdd2.
(Reading database ... 18923 files and directories currently installed.)
Preparing to unpack .../wsdd2_1.8.7+dfsg-1.1_arm64.deb ...
Unpacking wsdd2 (1.8.7+dfsg-1.1) ...
Setting up wsdd2 (1.8.7+dfsg-1.1) ...
Created symlink /etc/systemd/system/multi-user.target.wants/wsdd2.service → /lib/systemd/system/wsdd2.service.
root@DietPiOPi5:~#
```

```
root@DietPiOPi5:~# journalctl -u wsdd2.service
Aug 01 09:54:23 DietPiOPi5 systemd[1]: Started wsdd2.service - WSD/LLMNR Discovery/Name Service Daemon.
Aug 01 09:54:23 DietPiOPi5 wsdd2[2869]: cannot read additional dns hostnames from testparm
Aug 01 09:54:23 DietPiOPi5 wsdd2[2869]: cannot read netbios aliases from testparm
Aug 01 09:54:23 DietPiOPi5 wsdd2[2869]: starting.
root@DietPiOPi5:~#
```

As soon as the service is up, SAMBA share is becoming available within Windows network overview.

```
root@DietPiOPi5:~# systemctl status wsdd2.service
● wsdd2.service - WSD/LLMNR Discovery/Name Service Daemon
     Loaded: loaded (/lib/systemd/system/wsdd2.service; enabled; preset: enabled)
     Active: active (running) since Tue 2023-08-01 09:54:23 CEST; 57s ago
   Main PID: 2869 (wsdd2)
      Tasks: 1 (limit: 9201)
     Memory: 284.0K
        CPU: 353ms
     CGroup: /system.slice/wsdd2.service
             └─2869 /usr/sbin/wsdd2

Aug 01 09:54:23 DietPiOPi5 systemd[1]: Started wsdd2.service - WSD/LLMNR Discovery/Name Service Daemon.
Aug 01 09:54:23 DietPiOPi5 wsdd2[2869]: cannot read additional dns hostnames from testparm
Aug 01 09:54:23 DietPiOPi5 wsdd2[2869]: cannot read netbios aliases from testparm
Aug 01 09:54:23 DietPiOPi5 wsdd2[2869]: starting.
root@DietPiOPi5:~#
```

Memory footprint around 300K, probably we could install WSDD2 automatically to avoid to many user dialog and complexity?

**Reply by MichaIng** [reply]
> Memory footprint around 300K, probably we could install WSDD2 automatically to avoid to many user dialog and complexity?

Hmm, I am not big fans of installing additional services OOTB if not needed. I mean it is a convenience thing, only for Windows, and also only in some circumstances? One can always manually add the Samba share. There might be also cases where one explicitly does not want the SMB share to be broadcasted.

Strange output about `testparm`. It looks like errors, but actually it is just an info that additional hostnames and aliases just not provided in the Samba config, which is totally fine. Not sure why this should be logged at all: https://www.samba.org/samba/docs/current/man-html/testparm.1.html

**Comment by StephanStS** [comment]
@MichaIng : If we do not want to have wsdd2 as a selectable option within the samba package installation, we could close this discussion.
We described the wsdd2 installation there: https://dietpi.com/docs/software/file_servers/#samba-use-wsdd-to-view-shares

**Reply by MichaIng** [reply]
Let's keep it as is until first release next year. When dropping Bullseye support, it is only `apt install wsdd2`, hence easier to implement into `dietpi-software`. And then we can remove the section from the docs.

**Comment by StephanStS** [comment]
Discussion will be closed here. 
The installation of wsdd(2) is very easy and described there: https://dietpi.com/docs/software/file_servers/#samba-use-wsdd-to-view-shares
An extra option in `dietpi-software` therefore is unneeded in our view.
